akd_local_auditor/src/storage/s3/test.rs

```diff
@@ -308,14 +245,69 @@ async fn integration_test_audit_verification() -> Result<()> {
     Ok(())
 }
 
+// ============================ IMPORTANT ============================ //
+// NEVER run this operation in a CI pipeline, it takes 40+ min to complete
+// on reasonable hardware, and is only used for manual testing & timing information
+#[tokio::test]
+#[ignore]
+async fn expensive_audit_verification() -> Result<()> {
+    // make sure we have a valid bucket name, that's "somewhat" unique
+    let bucket = crate::common_test::alphanumeric_function_name!();
+    log::debug!("Test bucket name is {}", bucket);
+    assert!(matches!(validate_bucket_name(&bucket), Ok(_)));
+
+    // Get the storage reader
+    let storage = get_s3_storage(&bucket);
+    let shared_config = storage.get_shared_config().await;
+
+    // Populate the test storage
+    populate_test_storage(&shared_config, &bucket, 3, true).await?;
+
+    // List the epochs found in the storage layer
+    let mut epoch_summaries: Vec<EpochSummary> =
+        storage.list_proofs(ProofIndexCacheOption::NoCache).await?;
+    epoch_summaries.sort_by(|a, b| a.name.epoch.cmp(&b.name.epoch));
+
+    // There should be 3 proofs in the storage layer
+    log::info!(
+        "There are {} epochs in the storage layer",
+        epoch_summaries.len()
+    );
+    assert_eq!(3, epoch_summaries.len());
+
+    // Verify just 1 epoch
+    let now = std::time::Instant::now();
+
+    let epoch = &epoch_summaries[3];
+    let proof_blob = storage.get_proof(epoch).await?;
+    log::info!(
+        "Verification epoch {} -> {}",
+        epoch.name.epoch,
+        epoch.name.epoch + 1
+    );
+    crate::auditor::audit_epoch::<Hasher>(proof_blob.clone(), false).await?;
+    crate::auditor::audit_epoch::<Hasher>(proof_blob, true).await?;
+
+    // it prints '2'
+    log::error!(
+        "**** ELAPSED AUDITING TIME **** {}",
+        now.elapsed().as_secs()
+    );
+
+    // if the test is successful, try a cleanup of the storage now
+    // maybe_flush_storage(&shared_config, &bucket).await?;
+
+    Ok(())
+}
+
 #[tokio::test]
 #[ignore]
 async fn populate_test_bucket() -> Result<()> {
     // Populates the test bucket for use with the command-line REPL via the command
     // cargo run -p akd_local_auditor -- s3 --bucket populatetestbucket --region us-east-2 --endpoint http://127.0.0.1:9000 --access-key test --secret-key someLongAccessKey
 
     // make sure we have a valid bucket name, that's "somewhat" unique
-    let bucket = bucket_name!();
+    let bucket = crate::common_test::alphanumeric_function_name!();
     log::debug!("Test bucket name is {}", bucket);
     assert!(matches!(validate_bucket_name(&bucket), Ok(_)));
```

Comment:
I am a bit lost on the bucket name logic here... this generation and validation logic is duplicated in many places. I think we should have tests for the generation in a separate function and assume it works correctly when we use it.

Can you provide a bit more context on these bucket names?

Reply:
So these are S3 storage bucket names. To guarantee parallel tests don't conflict with each other, each test uses a different bucket for its tests which is derived from the function name. However buckets cannot contain the "_" char and must be capped @ 63 chars. It's only for generating independent test cases